README.md

```diff
@@ -113,38 +114,39 @@ module.exports = function(environment) {
 
 ```hbs
 {{! app/templates/login.hbs }}
-{{#if session.isWorking}}
+{{#if this.session.isWorking}}
   One sec while we get you signed in...
 {{else}}
   {{error}}
-  <a href="#" {{action 'signInViaFacebook'}}>
+  <a href="#" {{on "click" this.signInViaFacebook}}>
     Sign In with Facebook
   </a>
 {{/if}}
 ```
 
 ```JavaScript
 // app/routes/login.js
-export default Ember.Route.extend({
-  actions: {
-    signInViaFacebook: function(){
-      var route = this,
-          controller = this.controllerFor('login');
-      // The provider name is passed to `open`
-      this.get('session').open('facebook-oauth2').then(function(){
-        route.transitionTo('dashboard');
-      }, function(error){
-        controller.set('error', 'Could not sign you in: '+error.message);
-      });
-    }
+export default class LoginRoute extends Route {
+  @service session;
+
+  @action
+  signInViaFacebook() {
+    var route = this,
+        controller = this.controllerFor('login');
+    // The provider name is passed to `open`
+    this.session.open('facebook-oauth2').then(function(){
+      route.transitionTo('dashboard');
+    }, function(error){
+      controller.set('error', 'Could not sign you in: '+error.message);
+    });
   }
-});
+}
 ```
 
 ```JavaScript
 // app/torii-adapters/application.js
-export default Ember.Object.extend({
-  open: function(authentication){
+export default class ApplicationAdapter extends EmberObject {
```

Comment:
Adapters and providers don't _technically_ need to extend `EmberObject` but this is a shortcut to allow them to be injected via ember's dependency injection.

Discussion in the ember discord [here ](https://discord.com/channels/480462759797063690/483601670685720591/791334489041141760)around registering / injecting classes properly linked to [this](https://github.com/NullVoxPopuli/ember-contextual-services/blob/master/addon/contextual-service.ts#L5) example which would allow for correct registration / injection without extending `EmberObject` but would require the following to be defined in user adapter / provider classes:

```JavaScript
static create(injections) {
  return new this(injections);
}

constructor(injections) {
  Object.assign(this, injections);
}
```
It seems like the documentation for [dependency injection](https://guides.emberjs.com/release/applications/dependency-injection/#toc_factory-registrations) is a bit out of date and does not mention using ES6 classes at all so not sure what the preferred path is between these options.

Reply:
Alternatively I guess this complexity could be dealt with in base provider / adapter classes in torii itself that users can be guided to extend from when making their own provider / adapter classes.